apis/dispatch.php

```diff
@@ -40,12 +31,34 @@
         }
     }
 }
+// Maintain site id for multi site compatibility.
+// token is a 32 character hash followed by hex encoded site id.
+if ($resource === "/api/auth") {
+    // Get a site id from initial log in authentication.
+    $data = (array)(json_decode(file_get_contents("php://input")));
+    $site = empty($data['site']) ? "default" : $data['site'];
+    $_GET['site'] = $site;
+} else {
+    if (strlen($_SERVER["HTTP_X_API_TOKEN"]) > 32) {
+        $token = str_split($_SERVER["HTTP_X_API_TOKEN"], 32);
+        $_SERVER["HTTP_X_API_TOKEN"] = $token[0]; // reset hash to further the adventure.
+        $_GET['site'] = hex2bin($token[1]); // site id
+    }
```

Comment:
do you need an else here to set the site to default (in case the token is size 32)?

Reply:
Really should not happen and will add throw a 401 or 302 there.